waspc/src/Wasp/Generator/NpmInstall.hs

```diff
@@ -0,0 +1,106 @@
+module Wasp.Generator.NpmInstall
```

Comment:
Why not put it under Generator.Setup.NpmInstall?

Reply:
Are you suggesting I create a `Generator.Setup` module first then? Whereas there's a `ServerGenerator.Setup` that doesn't exist for the `Generator` itself, and should I be creating a whole new subdirectory for this one module?

Reply:
Yes, so you would create `Generator/Setup/` directory. And then you would have module `Generator/Setup/NpmInstall.hs`. In Haskell directories directly reflect names of the modules and vice versa, so don't feel bad about creating them if that makes sense for a certain module.

I am suggesting this name because I think that NpmInstall is part of the Setup.

What is your reasoning for putting it next to Setup? It is merely in order to avoid creating a new directory, Geenrator/Setup/?

I didn't understand what you meant about ServerGenerator.Setup .

Reply:
My reason for creating a new file was previous PR feedback by you; it was in Setup.hs before.

I always hold off creating directories and detailed organization in general until I see a clear purpose for them.

I prefer flat over nested and I have good reasons for this; codebases often have premature nesting where people have a lot of directories to choose from and slot things in by type of code, as opposed to a module with unified purpose and boundaries. This can prevent refactorings from happening - you can see some of that in the duplication between WebApp.Setup and Server.Setup, some of which I have eliminated, some still exists.

In this case there are strong ties between NpmDependencies and NpmInstall. If I were to create a module I'd be more likely to create a Npm related one, and put a lot of this in it, along with what's in AppSpec.App.Dependency. I'd take a closer look at Wasp.NpmDependency as well, which seems to duplicate lots of this behavior. I'd investigate organizing a npm dependency management library together as a concept, though inevitably some of the things would lie across the app spec or generation boundary.